tink-docker/main.go

```diff
@@ -94,3 +96,33 @@ func DownloadFile(filepath string, url string) error {
 	_, err = io.Copy(out, resp.Body)
 	return err
 }
+
+func rebootWatch() {
+	fmt.Println("Starting Reboot Watcher")
+
+	// Forever loop
+	for {
+		if fileExists("/worker/reboot") {
+			//Build the command, and execute
+			cmd := exec.Command("/sbin/reboot")
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			err := cmd.Run()
+			if err != nil {
+				panic(err)
+			}
+			break
+		}
+		// Wait one second before looking for file
+		time.Sleep(time.Second)
```

Comment:
This is kind of inefficient, a loop. Could you use [fsnotify](https://godoc.org/github.com/fsnotify/fsnotify) to watch for changes in `/worker` and see if the `reboot` file is created?

Reply:
What will be wayyyy better

Reply:
For today i'm happy with this, I started with fsnotify and couldn't get it to work?